security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -839,12 +850,15 @@ func (sc *SecretCache) generateSecret(ctx context.Context, token string, connKey
 
 	// If token is jwt format, construct host name from jwt with format like spiffe://cluster.local/ns/foo/sa/sleep
 	// otherwise just use sdsrequest.resourceName as csr host name.
-	csrHostName, err := constructCSRHostName(sc.configOptions.TrustDomain, token)
-	if err != nil {
-		cacheLog.Warnf("%s failed to extract host name from jwt: %v, fallback to SDS request"+
-			" resource name: %s", logPrefix, err, connKey.ResourceName)
-		csrHostName = connKey.ResourceName
+	csrHostName := connKey.ResourceName
+	if sc.secOpts.CredFetcher.GetType() == security.K8S {
+		csrHostName, err = constructCSRHostName(sc.configOptions.TrustDomain, token)
+		if err != nil {
+			cacheLog.Warnf("%s failed to extract host name from jwt: %v, fallback to SDS request"+
```

Comment:
This log (and likely the code) is incorrect. If we hit an error, csrHostName will be `""` not `connKey.ResourceName` as indicated by the log and in the previous code

Reply:
The existing code is using resource name as the default if there is an error. In fact, this "CSRHostName" is not really used by CA, because CA always uses the credential sub claim instead of the host name in the CSR. I want to keep the current behavior for now to be safe. In the future, we may be able to simply the code here further.

Reply:
You are not keeping the current behavior - if there is an error, now we return `""`, but previously we return resource name. If we don't use it lets remove (in another PR maybe, to keep diff small)

Reply:
Yup. you are right. Fix it to make sure it keeps the existing behavior. I will address it in another PR.